modules/firelens_log_config/main.tf

```diff
@@ -0,0 +1,136 @@
+
+locals {
+  tld = var.region == "eu-central-1" ? "eu" : "com"
+
+  ecs_shared_env_vars = <<EOF
+        { "name" : "ENVIRONMENT", "value" : "${var.environment_name}" },
+        { "name" : "APP_NAME", "value" : "${var.app_name}" }
+EOF
+
+  fire_lens_container = <<EOF
+  {
+    "essential": true,
+    "image": "906394416424.dkr.ecr.${var.region}.amazonaws.com/aws-for-fluent-bit:stable",
+    "name": "log_router",
+    "cpu": 0,
+    "user": "0",
+    "environment": [],
+    "volumesFrom": [],
+    "portMappings": [],
+    "mountPoints": [],
+    "firelensConfiguration": {
+      "type": "fluentbit",
+      "options": {
+        "enable-ecs-log-metadata": "true"
+      }
+    },
+    "memoryReservation": 50
+  }
+EOF
+
+datadog_ecs_agent_task_def = <<EOF
+{
+  "name": "datadog-agent",
+  "image": "${var.datadog_image}",
+  "essential": false,
+  "memoryReservation": 256,
+  "cpu": 10,
+  "mountPoints": [],
+  "volumesFrom": [],
+  "portMappings": [
+    {
+      "containerPort": 8125,
+      "hostPort": 8125,
+      "protocol": "udp"
+    },
+    {
+      "containerPort": 8126,
+      "hostPort": 8126,
+      "protocol": "tcp"
+    }
+  ],
+  "environment": [
+    {
+      "name": "ECS_FARGATE",
+      "value": "true"
+    },
+    {
+      "name": "DD_LOG_LEVEL",
+      "value": "warn"
+    },
+    {
+      "name": "DD_APM_ENABLED",
+      "value": "true"
+    },
+    {
+      "name": "DD_APM_NON_LOCAL_TRAFFIC",
+      "value": "true"
+    },
+    {
+      "name": "DD_SYSTEM_PROBE_ENABLED",
+      "value": "true"
+    },
+    {
+      "name": "DD_PROCESS_AGENT_ENABLED",
+      "value": "true"
+    },
+    {
+      "name": "DD_HEALTH_PORT",
+      "value": "5555"
+    },
+    {
+      "name": "DD_APM_RECEIVER_PORT",
+      "value": "8126"
+    },
+    {
+      "name": "DD_DOGSTATSD_NON_LOCAL_TRAFFIC",
+      "value": "true"
+    },
+    {
+      "name": "DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL",
+      "value": "true"
+    },
+    {
+      "name": "DD_DOGSTATSD_PORT",
+      "value": "8125"
+    },
+    {
+      "name": "DD_DOCKER_LABELS_AS_TAGS",
+      "value": "${replace(jsonencode(var.tags), "\"", "\\\"")}"
+    }
+  ],
+  "secrets": [
+    {
+      "name": "DD_API_KEY",
+      "valueFrom": "${aws_ssm_parameter.datadog_key.arn}"
+    }
+  ],
+  ${local.log_configuration}
+}
+EOF
+
+  log_configuration = <<EOF
+  "mountPoints": [],
+  "volumesFrom": [],
+  "logConfiguration": {
+        "logDriver": "awsfirelens",
+        "options": {
+          "Name": "datadog",
+          "compress": "gzip",
+          "Host": "http-intake.logs.datadoghq.${local.tld}",
+          "dd_service": "${var.app_name}",
+          "dd_source": "${var.dd_source}",
+          "dd_message_key": "log",
+          "dd_tags": "env:${var.environment_name},application:${var.app_name}-${var.environment_name},version:${var.datadog_image},task:${var.task_name}",
+          "TLS": "on",
+          "provider": "ecs"
+        },
```

Comment:
Do we need secretOptions here? Just looking at https://github.com/smartrent/terraform-aws-ecs-container_definitions/blob/eefdc7d52f1d43621c9380a6df266d66fc444ffa/firelens_log_config/main.tf#L20

Reply:
Ah, I meant to ask about this:

In the [PR](https://github.com/smartrent/identity/commit/30e9d928996472097ed3e67c7fa3aa11dfb03664) you recommended I look at, there was the the secret options in the logConfig:
<img width="803" alt="Screen Shot 2022-04-28 at 6 26 19 PM" src="https://user-images.githubusercontent.com/21345512/165863518-c6066e81-77af-4e98-9461-c6811387ac6d.png">

As well as one in the Datadog container, but they are both datadog api keys:
<img width="486" alt="Screen Shot 2022-04-28 at 6 26 40 PM" src="https://user-images.githubusercontent.com/21345512/165863808-16dd5b8a-cb37-432a-a2a9-19a2864b2002.png">

So I put it in the datadog task definition and have the key set [in this line](https://github.com/smartrent/nerves-hub-terraform/pull/30/files/2ee92980a959be10135f59b069d0df2d0ab085b1#diff-f275be9fd58c428c299725380fb9b144a187f2f678014da65f86108b50da97e8R109). And when I look at the containers in dev, the nerves-hub-dev and the datadog container both have access to the Datadog API Key. I'm not positive this should be that way, but I did want to get your opinion.  The Log Configuration is also part of the Datadog task definition as I have it currently set up.

Reply:
Ah gotcha. Yeah we'll need the api key in both the Datadog task def and also in the firelens config (double checked the [datadog docs](https://docs.datadoghq.com/integrations/ecs_fargate/?tab=fluentbitandfirelens) since I see what you mean and I wasn't sure either 😬). The repo I linked has our latest and greatest configs (the PR I previously recommended is kinda outdated 😬.

Reply:
Also, we have the datadog api key in both the log configuration and the datadog log configuration, so I _think_ this is good?

Reply:
This looks good. Thanks!